cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -581,7 +581,9 @@ func CreateControllerContext(ctx context.Context, s *config.CompletedConfig, roo
 	// Informer transform to trim ManagedFields for memory efficiency.
 	trim := func(obj interface{}) (interface{}, error) {
 		if accessor, err := meta.Accessor(obj); err == nil {
-			accessor.SetManagedFields(nil)
+			if accessor.GetManagedFields() != nil {
```

Comment:
GetManagedFields() is very cheap on the `ObjectMeta#GetManagedFields()` implementation and very expensive on the `Unstructured#GetManagedFields()` implementation ... can you double check that this is only used on typed objects and not unstructured?